scio-core/src/main/scala/com/spotify/scio/util/Functions.scala

```diff
@@ -197,4 +199,12 @@ private[scio] object Functions {
     override def reduceOption(accumulator: Iterable[T]): Option[T] =
       _mon.sumOption(accumulator).orElse(Some(_mon.zero))
   }
+
+  def memoize[T, U](f: T => U): T => U = {
+    lazy val cache = new ConcurrentHashMap[T, U]()
```

Comment:
How many `cache` instances will this produce? Will it become a member of `object Functions`?

Reply:
this will produce one `cache` instance per function. I don't believe so.